python/lsst/ctrl/bps/parsl/sites/princeton.py

```diff
@@ -104,4 +105,8 @@ def get_address(self) -> str:
         interface, because the cluster nodes can't connect to the head node
         through the regular internet.
         """
-        return address_by_interface("ib0")
+        net_interfaces = [interface for interface in net_if_addrs().keys() if interface[:2] in ["ib", "op"]]
+        if net_interfaces:
+            return address_by_interface(net_interfaces[0])
+        else:
```

Comment:
Drop the `else`.

Reply:
This function may then return nothing, which would cause `make_executor` inside `lsst/ctrl/bps/parsl/sites/slurm.py` to fail with a generic error. Is that better than a specific error raised here?

Reply:
I interpreted @PaulPrice as saying write:

```python
if net_interfaces:
    return blah
raise RuntimError("reason")
```

Reply:
Ah, literally just the `else` line, thanks. Yes, that makes sense, cheers.